x-pack/plugin/ent-search/qa/rest/src/yamlRestTest/resources/rest-api-spec/test/entsearch/rules/50_query_rule_put.yml

```diff
@@ -0,0 +1,261 @@
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ elastic ]
+          actions:
+            ids:
+              - 'id1'
+              - 'id2'
+
+  # Finally, add another single rule with no priority. This should be appended to the ruleset.
+  - do:
+      query_rule.put:
+        ruleset_id: test-ruleset
+        rule_id: query-rule-id4
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'search' ]
+          actions:
+            ids:
+              - 'id2'
+
+  - match: { result: 'updated' }
+
+  - do:
+      query_ruleset.get:
+        ruleset_id: test-ruleset
+
+  - match: { ruleset_id: test-ruleset }
+  - match:
+      rules:
+        - rule_id: query-rule-id2
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ kibana ]
+          actions:
+            ids:
+              - 'id3'
+              - 'id4'
+          priority: 1
+        - rule_id: query-rule-id3
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ logstash ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 2
+        - rule_id: query-rule-id1
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ elastic ]
+          actions:
+            ids:
+              - 'id1'
+              - 'id2'
+        - rule_id: query-rule-id4
+          type: pinned
+          criteria:
+            - type: exact
+              metadata: query_string
+              values: [ search ]
+          actions:
+            ids:
+              - 'id2'
+
+
+---
+'Create Query Rule - Resource already exists':
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id1'
+          priority: 5
+
+  - match: { result: 'created' }
+
+  - do:
+      query_rule.put:
+        ruleset_id: test-query-rule-recreating
+        rule_id: abc
+        body:
+          type: 'pinned'
+          criteria:
+            type: 'exact'
+            metadata: 'query_string'
+            values: [ 'elastic' ]
+          actions:
+            ids:
+              - 'id2'
+          priority: 3
+
+  - match: { result: 'updated' }
+
+---
+'Create Query Rule - Insufficient privilege':
+  - skip:
+      features: headers
+
+  - do:
+      catch: forbidden
+      headers: { Authorization: "Basic ZW50c2VhcmNoLXVzZXI6ZW50c2VhcmNoLXVzZXItcGFzc3dvcmQ=" }  # user
+      query_rule.put:
+        ruleset_id: forbidden-query-ruleset
```

Comment:
Should we cleanup this ruleset in the `teardown` section, just in case this test fails and it is created?

Reply:
Can't hurt